### PR TITLE
Show connect buttons for installed apps only

### PR DESCRIPTION
**Pull request by humitos**
Instead of hardcode the button/form for each of the services, we share
the ``provider_list.html`` file used in the "Connected Services" tab
from "Account Settings".

This template shows only buttons for the Services installed in the
database.

> Pay attention to the CSS changes since I probable screwed something :)

### CODE REVIEW

**Comment by agjohnson** [comment]
Could you post some screenshots of the two changes?

**Comment by humitos** [comment]
Import project

captura de pantalla_2017-12-11_15-26-07

Settings page

captura de pantalla_2017-12-11_15-26-20

File: media/css/core.css [hunk 1]
```diff
@@ -690,19 +690,16 @@ div.project-import-remote button.remote-sync:before {
...
-div.project-import-remote form.import-connect-github button:before,
 a.socialaccount-provider.github:before {
     font-family: FontAwesome;
     content: "\f09b";
 }
 
-div.project-import-remote form.import-connect-gitlab button:before,
 a.socialaccount-provider.gitlab:before {
     font-family: FontAwesome;
     content: "\f1d3";
```
Comment (safwanrahman) [inline media/css/core.css]: `\f296` should be for gitlab
http://fontawesome.io/icon/gitlab/

File: readthedocs/restapi/views/footer_views.py [hunk 2]
```diff
@@ -43,7 +43,7 @@ def get_version_compare_data(project, base_version=None):
     }
     if highest_version_obj:
         ret_val['url'] = highest_version_obj.get_absolute_url()
-        ret_val['slug'] = highest_version_obj.slug,
+        ret_val['slug'] = highest_version_obj.slug
```
Comment (ericholscher) [inline readthedocs/restapi/views/footer_views.py]: This looks like an unrelated bug?

Comment (humitos) [reply readthedocs/restapi/views/footer_views.py]: Ha! Yeah. I don't understand how this passed the test in the PR that was merged :/

File: readthedocs/templates/projects/project_import.html [hunk 3]
```diff
@@ -93,41 +93,12 @@ <h1>{% trans "Import a Repository" %}</h1>
...
-                          class="import-connect import-connect-bitbucket">
-                        <button>
-                          {% trans "Connect to Bitbucket" %}
-                        </button>
-                        <input type="hidden" name="process" value="connect">
-                        <input type="hidden" name="next" value="/dashboard/import/">
-                      </form>
+                    <div class="project-import-providers">
+                      <ul class="socialaccount_providers">
+                        {% include "socialaccount/snippets/provider_list.html" with process="connect" %}
```
Comment (ericholscher) [inline readthedocs/templates/projects/project_import.html]: I don't see this file in the commit, did it already exist?

Comment (ericholscher) [reply readthedocs/templates/projects/project_import.html]: Oh I see it below -- is it also used somewhere else?

File: media/css/core.css (hunk 1 above)
Comment (humitos) [reply media/css/core.css]: Nice!

File: readthedocs/templates/projects/project_import.html (hunk 3 above)
Comment (humitos) [reply readthedocs/templates/projects/project_import.html]: Yes, it is used in the `/accounts/social/connections/` page, under the admin settings.

Comment (agjohnson) [reply readthedocs/templates/projects/project_import.html]: We do lose the `next` parameter here, correct? This will likely just redirect users back to the connection page. Is redirecting to the import page important here?

File: readthedocs/projects/static/projects/css/import.css [hunk 4]
```diff
@@ -49,12 +49,14 @@ div.project-import-remote li.remote-repo ul.remote-repo-menu {
...
+div.project-import-remote div.project-import-providers ul {
+  overflow: hidden;
+}
+div.project-import-remote div.project-import-providers ul li {
+  float: left;
 }
-div.project-import-remote form.import-connect button {
-  display: inline;
+div.project-import-remote a.socialaccount-provider {
+  display: block;
```
Comment (agjohnson) [inline readthedocs/projects/static/projects/css/import.css]: These changes look to have some negative side effect on spacing and alignment. What prompted these changes? If we have to alter layout on one of the forms, that might be better.

File: media/css/core.css [hunk 5]
```diff
@@ -690,19 +690,16 @@ div.project-import-remote button.remote-sync:before {
...
 }
 
-div.project-import-remote form.import-connect-bitbucket button:before,
```
[the remark this answers is not in the document]
Comment (humitos) [reply media/css/core.css]: I removed these because the `form` is not used anymore (I removed them). So, I didn't want to leave orphans styles...

File: readthedocs/projects/static/projects/css/import.css (hunk 4 above)
Comment (humitos) [reply readthedocs/projects/static/projects/css/import.css]: "I don't know CSS", so I don't have _good reasons_. I touched it a little, googled something and it looked it worked so I thought that I could propose it.

I'm totally 👍 on remove this and change it as you like/prefer/wish/etc.

File: readthedocs/templates/projects/project_import.html (hunk 3 above)
Comment (humitos) [reply readthedocs/templates/projects/project_import.html]: Good catch! I didn't realize this.

Yes, I think it's important since you are at the import page. I'd like to connect my account and import the my project immediately. I'm proposing a fix for this using the template tag: https://github.com/pennersr/django-allauth/blob/cfc5112bf474af2cbb16e4b4a36708e8949a2105/allauth/socialaccount/templatetags/socialaccount.py#L45

File: readthedocs/projects/static/projects/css/import.css (hunk 4 above)
Comment (agjohnson) [reply readthedocs/projects/static/projects/css/import.css]: Also, missed this the first time. This is actually editing a compiled css file, the source is at `readthedocs/projects/static-src/projects/css/import.less`. I'm going to be working to unify our styling, and to get rid of our `core.css` file soon.

**Comment by agjohnson** [comment]
Updated the styles for the import list page:

image
